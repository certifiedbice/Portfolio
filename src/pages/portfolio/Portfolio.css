.gallery{
	flex: 1 calc(25%);
	align-items: stretch;
}
@media(min-width:700px){
.gallery{
	max-width: 32.5%;
}
}
.item-status{
	margin: 20px;
}
.item-status span{
	font-weight: 700;
}
.item-stack{
	display: flex;
	flex-wrap: wrap;
	justify-content: flex-start;
	margin: 20px;
}
.item-stack-title{
	margin: 5px;
	font-weight: 700;
}
.item-stack-item{
	margin: 5px;
}
.item-description{
	line-height: 2;
}