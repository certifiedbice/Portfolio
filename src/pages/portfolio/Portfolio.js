import React,{Component} from 'react';
import Context from '../../masterProvider';
import DataHelper from '../../dataHelper';
import Header from '../../components/header/Header';
import Footer from '../../components/footer/Footer';
import Fade from '../../components/fade/Fade';
import Background from '../../components/background/Background';
import './Portfolio.css';
export default class Portfolio extends Component{
	static contextType=Context;
	componentDidMount(){
		// Compensate for content outside the viewport.
		this.context.verticalPadAndScroll();
	}
	render(){
		const web=DataHelper.web;
		const art=DataHelper.art;
		return(
			<>
				<Background/>
				<main>
					<section id='portfolio'>
						<div className='group'>
							<div className='item'>
								<h3>Web Design &#38; Development</h3>
							</div>
						</div>
						<div className='group'>
							{
								web.map(item=>{
									if(item.link!==null){
										return (
											<div key={item.title} className='item gallery'>
												<a href={item.link} target='_blank' rel='noopener noreferrer'>
													<img src={require(`../../images/galleries/web/${item.image}`)} alt={item.title}/>
												</a>
												<h3>{item.title}</h3>
												<div className='item-status'><span>Status:</span> {item.status}</div>
												<div className='item-stack'>
													<div className='item-stack-title'>Stack: </div>
													{
														item.stack.map(item=>{
															return(
																<div key={item} className='item-stack-item'>{item},</div>
															);
														})
													}
												</div>
												<div className='item-description'>{item.description}</div>
											</div>
										);
									}
									else return null;
								})
							}
						</div>  
						<div className='group'>
							<div className='item'>
								<h3>Art &#38; Design</h3>
							</div>
						</div>
						<div className='group'>
							{
								art.map(item=>{
									if(item.link!==null){
										return (
											<div key={item.title} className='item gallery'>
												<a href={require(`../../images/galleries/video/${item.link}`)} target='_blank' rel='noopener noreferrer'>
													<img src={require(`../../images/galleries/art/${item.image}`)} alt={item.title}/>
												</a>
												<h3>{item.title}</h3>
												<div className='item-description'>{item.description}</div>
											</div>
										);
									}
									else{
										return (
											<div key={item.title} className='item gallery'>
												<img src={require(`../../images/galleries/art/${item.image}`)} alt={item.title}/>
												<h3>{item.title}</h3>
												<div className='item-description'>{item.description}</div>
											</div>
										);
									}
								})
							}
						</div>
					</section>
				</main>
				<Header/>
				<Footer/>
				<Fade location={this.props.location.pathname}/>
			</>
		);
	}
}