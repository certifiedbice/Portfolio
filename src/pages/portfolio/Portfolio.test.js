import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Portfolio from './Portfolio';

it('renders without crashing',()=>{
	const pathname={location:{pathname:'/portfolio'}};
    const div=document.createElement('div');
    ReactDOM.render(<BrowserRouter><Portfolio location={pathname}/></BrowserRouter>,div);
});

it('renders correctly',()=>{
	const pathname={location:{pathname:'/portfolio'}};
    const tree=renderer
        .create(<BrowserRouter><Portfolio location={pathname}/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});