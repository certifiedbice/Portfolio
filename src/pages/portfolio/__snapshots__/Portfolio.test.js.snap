// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    id="background-image-container"
  >
    <div
      id="background-image"
    />
  </div>,
  <main>
    <section
      id="portfolio"
    >
      <div
        className="group"
      >
        <div
          className="item"
        >
          <h3>
            Web Design & Development
          </h3>
        </div>
      </div>
      <div
        className="group"
      >
        <div
          className="item gallery"
        >
          <a
            href="http://felonydiscrimination.org.jpg"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="felonydiscrimination.org"
              src="felonydiscrimination.org.png"
            />
          </a>
          <h3>
            felonydiscrimination.org
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live beta testing
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              React
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Node
              ,
            </div>
            <div
              className="item-stack-item"
            >
              PostgreSQL
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Express
              ,
            </div>
            <div
              className="item-stack-item"
            >
              JSON
              ,
            </div>
            <div
              className="item-stack-item"
            >
              JWT
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Knex
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Cors
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Dotenv
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Morgan
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Helmet
              ,
            </div>
            <div
              className="item-stack-item"
            >
              XSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Mocha
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Chai
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Enzyme
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Jest
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Supertest
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            Combating the rampant discrimination against persons with a felony history. This app provides a service to catalog, rate, and comment on companies and organizations in the areas of housing and employment that discriminate soley on the basis of a person having a felony in their background.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="http://camasforcannabis.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="camasforcannabis.com"
              src="camasforcannabis.com.png"
            />
          </a>
          <h3>
            camasforcannabis.com
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              PHP
              ,
            </div>
            <div
              className="item-stack-item"
            >
              WordPress
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            Camas For Cannabis Access, Initiative Measure No. 1. The app was built to serve as a resource to the organization so they could provide useful information to help people in Camas make an informed decision when voting on this ballot measure.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="http://certifiedbice.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="certifiedbice.com"
              src="certifiedbice.com.png"
            />
          </a>
          <h3>
            certifiedbice.com
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Under Development
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            The platform, incubator, and launchpad of Bice. This serves as an online presence for my entreprenurial endeavors.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="http://bice.blog"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="bice.blog"
              src="bice.blog.png"
            />
          </a>
          <h3>
            bice.blog
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              PHP
              ,
            </div>
            <div
              className="item-stack-item"
            >
              WordPress
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            The blog of Bice. A basic WordPress in which I am blogging when I have time.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="http://bice.rocks"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="bice.rocks"
              src="bice.rocks.png"
            />
          </a>
          <h3>
            bice.rocks
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              React
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Enzyme
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Jest
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            The professional portfolio of Bice. I built this for the purpose of showcasing my various talents and skills.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="https://github.com/certifiedbice/Sudoku"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Sudoku"
              src="sudoku.bice.rocks.png"
            />
          </a>
          <h3>
            Sudoku
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Under Development
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            Vanilla Javascript Sudoku Generator. Originally built with no internet access and no resources or research material other than a basic JavaScript reference manual. I built the algorithm from scratch, I was forced to record my fully function code onto notebook paper and am in the process of re-typing and fixing the errors from documenting it in that fashion and hope to have this up very soon.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="https://github.com/certifiedbice/petful-client"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Petful Pet Adoption App"
              src="petful.png"
            />
          </a>
          <h3>
            Petful Pet Adoption App
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live Demo
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Reactjs
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Nodejs
              ,
            </div>
            <div
              className="item-stack-item"
            >
              PostgreSQL
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Heroku
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Vercel
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            FIFO pet adoption app for demonstration purposes. A paired exercise completed by a &lt;team member&gt; and myself, this is a functional demonstration of a queued system that is first-in, first-out.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="https://github.com/certifiedbice/spaced-repetition-client"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Spaced Repetition Language Learning App"
              src="spaced-repetition.png"
            />
          </a>
          <h3>
            Spaced Repetition Language Learning App
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live Demo
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Reactjs
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Nodejs
              ,
            </div>
            <div
              className="item-stack-item"
            >
              PostgreSQL
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Heroku
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Vercel
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            Singly linked list spaced repetition learning app for demonstration purposes. This was a paired project in which I worked closely with a team member to create this app. Although a variety of difficulties arose around this project we were able to complete a fully functional product.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="https://buddyup-client.vercel.app/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="BuddyUp"
              src="buddyup.png"
            />
          </a>
          <h3>
            BuddyUp
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Live Demo
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Reactjs
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Nodejs
              ,
            </div>
            <div
              className="item-stack-item"
            >
              PostgreSQL
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Heroku
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Vercel
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            An app for making friends. This was a 5 man team effort, myself and another team member completed the backend system. While the other 3 completed the frontend. I was personally responsible for the initial boilerplate API, the user authentication system, and all backend testing.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="http://code-examples.bice.rocks"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Code Examples"
              src="code-examples.bice.rocks.png"
            />
          </a>
          <h3>
            Code Examples
          </h3>
          <div
            className="item-status"
          >
            <span>
              Status:
            </span>
             
            Under Development
          </div>
          <div
            className="item-stack"
          >
            <div
              className="item-stack-title"
            >
              Stack: 
            </div>
            <div
              className="item-stack-item"
            >
              HTML5
              ,
            </div>
            <div
              className="item-stack-item"
            >
              CSS
              ,
            </div>
            <div
              className="item-stack-item"
            >
              Javascript
              ,
            </div>
          </div>
          <div
            className="item-description"
          >
            Various code examples of my work for demonstration purposes.
          </div>
        </div>
      </div>
      <div
        className="group"
      >
        <div
          className="item"
        >
          <h3>
            Art & Design
          </h3>
        </div>
      </div>
      <div
        className="group"
      >
        <div
          className="item gallery"
        >
          <img
            alt="Fuegon Spirits Poster 01"
            src="fuegon-poster-01.jpg"
          />
          <h3>
            Fuegon Spirits Poster 01
          </h3>
          <div
            className="item-description"
          >
            Graphic Design employing a 3D model for the purposes of marketing material.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <img
            alt="Fuegon Spirits Poster 02"
            src="fuegon-poster-02.jpg"
          />
          <h3>
            Fuegon Spirits Poster 02
          </h3>
          <div
            className="item-description"
          >
            Graphic Design employing a 3D model for the purposes of marketing material.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <a
            href="fuegon-commercial.mp4"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Fuegon Spirits Commercial"
              src="fuegon-commercial.jpg"
            />
          </a>
          <h3>
            Fuegon Spirits Commercial
          </h3>
          <div
            className="item-description"
          >
            A complete work including Graphic Design, 3D Modeling & Animation, Video & Audio Production for the purposes of marketing material.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <img
            alt="Hang in there baby"
            src="hang-in-there-baby.jpg"
          />
          <h3>
            Hang in there baby
          </h3>
          <div
            className="item-description"
          >
            Digital illustration using Photoshop & Illustrator.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <img
            alt="Porpkist Dolphin"
            src="porpkist-dolphin.jpg"
          />
          <h3>
            Porpkist Dolphin
          </h3>
          <div
            className="item-description"
          >
            Digital illustration using Illustrator.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <img
            alt="Unknown Female"
            src="unknown-female-figure-2019.jpg"
          />
          <h3>
            Unknown Female
          </h3>
          <div
            className="item-description"
          >
            Ball point pen on paper, preliminary concept for a series I will be completing in the future.
          </div>
        </div>
        <div
          className="item gallery"
        >
          <img
            alt="Angel on the steps"
            src="angel-on-the-steps-of-the-pantheon-2019.jpg"
          />
          <h3>
            Angel on the steps
          </h3>
          <div
            className="item-description"
          >
            Ball point pen on paper.
          </div>
        </div>
      </div>
    </section>
  </main>,
  <header
    id="site-header"
  >
    <div
      id="site-info"
    >
      <h1
        id="site-title"
      >
        Portfolio of Bice
      </h1>
      <div
        id="site-description"
      >
        My various projects in all their variousness.
      </div>
      <nav>
        <ul
          id="primary-menu"
        >
          <li>
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/portfolio"
            >
              Portfolio
            </a>
          </li>
          <li>
            <a
              href="/resume"
            >
              Resume
            </a>
          </li>
          <li>
            <a
              href="/services"
            >
              Services
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <footer
    id="site-footer"
  >
    Bice - 2020
    <div
      id="footer-links-container"
    >
      <a
        className="footer-social-icon-link"
        href="https://www.facebook.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Facebook"
          className="footer-social-icons-image"
          src="facebook.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://github.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="GitHub"
          className="footer-social-icons-image"
          src="github.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.instagram.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Instagram"
          className="footer-social-icons-image"
          src="instagram.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.hackerrank.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="HackerRank"
          className="footer-social-icons-image"
          src="hackerrank.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.linkedin.com/in/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="LinkedIn"
          className="footer-social-icons-image"
          src="linkedin.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://twitter.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Twitter"
          className="footer-social-icons-image"
          src="twitter.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.youtube.com/channel/UCEITFHDFDtvTIKFbUhG_E9w"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="YouTube"
          className="footer-social-icons-image"
          src="youtube.png"
        />
      </a>
    </div>
  </footer>,
  <div
    id="page-fade"
  />,
]
`;
